Fix 2.4.0 missing docs of cgo client (#6974)
apache/pulsar: site2/website/versioned_docs/version-2.4.0/client-libraries-go.md

}
```

### Client operations

> Pulsar Go client have the following methods available:

Reviewer: ```suggestion
Pulsar Go client has the following methods available:
```

Method | Description | Return type
:------|:------------|:-----------

> `CreateProducer(ProducerOptions)` | Create the producer instance(This method will block until the producer is created successfully). | `(Producer, error)`

Reviewer: ```suggestion
`CreateProducer(ProducerOptions)` | Create the producer instance. (This method will be blocked until the producer is created successfully.) | `(Producer, error)`
```

> `CreateProducerWithSchema(ProducerOptions, Schema)` | Create the producer instance with schema | `(Producer, error)`

Reviewer: ```suggestion
`CreateProducerWithSchema(ProducerOptions, Schema)` | Create a producer instance with schema. | `(Producer, error)`
```

`Subscribe(ConsumerOptions)` | Create a `Consumer` by subscribing to a topic. | `(Consumer, error)`
`SubscribeWithSchema(ConsumerOptions, Schema)` | Create a `Consumer` with schema by subscribing to a topic. | `(Consumer, error)`
`CreateReader(ReaderOptions)` | Create a Reader instance. | `(Reader, error)`
`CreateReaderWithSchema(ReaderOptions, Schema)` | Create a Reader instance with schema. | `(Reader, error)`
`TopicPartitions(topic string)` | Fetch the list of partitions for a given topic. | `([]string, error)`

> `Close()` | Close the Client and free associated resources. | `error`

Reviewer: ```suggestion
`Close()` | Close the Pulsar Go client and release associated resources. | `error`
```

`NackID(MessageID)` | Acknowledge the failure to process a single message. | `error`
`Close()` | Closes the consumer, disabling its ability to receive messages from the broker | `error`
`RedeliverUnackedMessages()` | Redelivers *all* unacknowledged messages on the topic. In [failover](concepts-messaging.md#failover) mode, this request is ignored if the consumer isn't active on the specified topic; in [shared](concepts-messaging.md#shared) mode, redelivered messages are distributed across all consumers connected to the topic. **Note**: this is a *non-blocking* operation that doesn't throw an error. |

> `Schema()` | Message schema definition | `Schema`

Reviewer: ```suggestion
`Schema()` | Set the message schema definition. | `Schema`
```

}
```

#### Schema example

> Here's an example usage of

Reviewer: it's better to use a whole sentence as the leading-in sentence. such as:

This example shows how to create a producer and consumer with schema.

`Name` | The name of the consumer |
`AckTimeout` | | 0
`NackRedeliveryDelay` | The delay after which to redeliver the messages that failed to be processed. Default is 1min. (See `Consumer.Nack()`) | 1 minute
`SubscriptionType` | Available options are `Exclusive`, `Shared`, `Key_Shared` and `Failover` | `Exclusive`
`MessageChannel` | The Go channel used by the consumer. Messages that arrive from the Pulsar topic(s) will be passed to this channel. |
`ReceiverQueueSize` | Sets the size of the consumer's receiver queue, i.e. the number of messages that can be accumulated by the consumer before the application calls `Receive`. A value higher than the default of 1000 could increase consumer throughput, though at the expense of more memory utilization. | 1000
`MaxTotalReceiverQueueSizeAcrossPartitions` |Set the max total receiver queue size across partitions. This setting will be used to reduce the receiver queue size for individual partitions if the total exceeds this value | 50000

> `Name` | Set the consumer name | `string`

Reviewer: ```suggestion
`Name` | Set the consumer name. | `string`
```

> `ReadCompacted` | If enabled, the consumer will read messages from the compacted topic rather than reading the full message backlog of the topic.| `bool`

Reviewer: ```suggestion
`ReadCompacted` | If this parameter is enabled, the consumer reads messages from the compacted topic rather than reading the full message backlog of the topic.| `bool`
```
